source/module_base/test/math_integral_test.cpp

```diff
@@ -0,0 +1,93 @@
+#include"../math_integral.h"
+#include"gtest/gtest.h"
+
+#include<math.h>
+
+#define PI 3.14159265358979
+#define doublethreshold 1e-12
+
+/************************************************
+*  unit test of class Integral
+***********************************************/
+
+/**
+ * Tested functions:
+ *  - function Simpson_Integral with rab as input
+ *  - function Simpson_Integral with dr as input
+ *  - function Simpson_Integral_0toall
+ *  - function Simpson_Integral_alltoinf
+ * 
+ */
+
+class Simpson_Integral_sinx : public testing::Test
+{
+    /**
+     * test the integral of sinx between [0,PI],
+     * devide to mesh-1 parts
+     */
+    
+    protected:
+
+    double*     func;
+    double*     rab;
+    int         mesh = 10001;
+    double      dr   = PI/(mesh-1);
+    double      asum;
+    double*     asumlist;
+    double      expectvalue = 2.0;
+
+    void SetUp()
+    {
+        func        = new double[mesh];
+        rab         = new double[mesh];
+        asumlist    = new double[mesh];
+
+        for (int i=0;i<mesh;i++)
+        {
+            func[i] = sin(PI*i/(mesh-1));
+            rab[i] = dr;
+        }
+    }
+
+    void TearDown()
+    {
+        delete func;
+        delete rab;
+        delete asumlist;
+    }
+};
+
+TEST_F(Simpson_Integral_sinx,Simpson_Integral_rab)
```

Comment:
Both arguments TestFixtureName and TestName must be valid C++ identifiers and must not contain underscores (_). See https://google.github.io/googletest/reference/testing.html#TEST_F . 
@pxlxingliang Your former PRs also have the same problem. Sorry for my ignorance, it's my fault. Would you help me fixing them in another PR and renew test document? I suggest upper camel case is better (e.g. "SimpsonIntegralSinx").

Reply:
ok, I will modify it. But I don't know the reason of not containing underscores, and It seems has no any impact on the previous using of unit test.

Reply:
gtest already has the explanation for this.
http://google.github.io/googletest/faq.html#why-should-test-suite-names-and-test-names-not-contain-underscore

Reply:
> ok, I will modify it. But I don't know the reason of not containing underscores, and It seems has no any impact on the previous using of unit test.

It's just conforms to C++ standard, and theoretically there would be no impact.